akka-cluster-typed/src/test/scala/docs/akka/cluster/typed/DistributedPubSubExample.scala

```diff
@@ -0,0 +1,355 @@
+
+        case RegistrationAck(status: RegistrationSuccess, onBehalfOf) =>
+          context.log.info(
+            "New data type was registered, provisioning destination for {}.",
+            context.self.path / status.dataType.key)
+
+          val ingestion = context.spawn(Ingestion(status.dataType, mediator), status.dataType.key)
+          ingestion ! StartIngestion(None, None) // stubbed source/sink from another service
+          onBehalfOf ! ProvisionSuccess(status.dataType.key)
+          Behaviors.same
+
+      }
+    }
+  }
+
+}
+
+object DataPlatform {
+  import Ontology._
+
+  def apply(): Behavior[ProvisionCommand] = {
+    Behaviors.setup { context =>
+      //#mediator
+      import akka.actor.typed.scaladsl.adapter._
+      val mediator = DistributedPubSub(context.system.toClassic).mediator
+      //#mediator
+      val service = context.spawn(DataService(mediator), "data")
+
+      Behaviors.receiveMessagePartial {
+        case CreateSubscriber(key) =>
+          // subscribe to provisioning for this data type
+          //#start-subscribers
+          context.spawnAnonymous(Subscriber(key, mediator))
+          //#start-subscribers
+          Behaviors.same
+
+        case cmd: ProvisionCommand =>
+          service ! cmd
+          Behaviors.same
+
+      }
+    }
+  }
+}
+
+object DistributedPubSubExample {
+  import akka.actor.testkit.typed.scaladsl.TestProbe
+  import Ontology._
+
+  val config: Config = ConfigFactory.parseString(s"""
+        akka.actor.provider = "cluster"
+        akka.cluster.pub-sub.max-delta-elements = 500
+        # sample vs test so this does not get registered?
+        akka.cluster.downing-provider-class = akka.cluster.testkit.AutoDowning
+        akka.cluster.jmx.enabled = off
+        akka.remote.artery.canonical.hostname = 127.0.0.1
+        akka.remote.classic.netty.tcp.port = 0
+        akka.remote.artery.canonical.port = 0
+        akka.loglevel = INFO
+        akka.loggers = ["akka.testkit.TestEventListener"]
+    """)
+
+  def createCluster(nodes: List[ActorSystem[_]]): Unit = {
+    val clusterUp = (nodes: List[ActorSystem[_]], expected: Int) =>
+      nodes.forall { s =>
+        Cluster(s).state.members.count(_.status == MemberStatus.up) == expected
+      }
+
+    val awaitClusterUp = (nodes: List[ActorSystem[_]], expected: Int) =>
+      while (!clusterUp(nodes, expected)) {
+        Thread.sleep(1000)
+      }
+
+    val seed = nodes.head
+    val joinTo = Cluster(seed).selfMember.address
+
+    Cluster(seed).manager ! Join(joinTo)
+    awaitClusterUp(List(seed), 1)
+    nodes.tail.foreach(Cluster(_).manager ! Join(joinTo))
+    awaitClusterUp(nodes, nodes.size)
+  }
+
+  def main(args: Array[String]): Unit = {
+
+    val system = ActorSystem[ProvisionCommand](DataPlatform(), "DataPlatform", config)
+    val system2 = ActorSystem[ProvisionCommand](DataPlatform(), system.name, config)
+    val system3 = ActorSystem[ProvisionCommand](DataPlatform(), system.name, config)
```

Comment:
back to a multi-j

Reply:
I don't think we should spend too much time on this, since we want to replace it soon anyway